core/src/main/java/org/akaza/openclinica/service/managestudy/DiscrepancyNoteService.java

```diff
@@ -54,6 +61,57 @@ private DiscrepancyNoteBean createDiscrepancyNoteBean(String description, int en
 
     }
 
+    /**
+     * 
+     * @param itemName
+     * @param message
+     * @param eventCrf
+     * @param itemData
+     * @param parentId
+     * @param uab
+     * @param ds
+     * @param studyId
+     * @param detailedNotes
+     * @param discrepancyNoteTypeId
+     * @return
+     */
+    public DiscrepancyNoteBean createDiscrepancyNote(String itemName, String message, EventCrf eventCrf, ItemData itemData,
+            Integer parentId, UserAccountBean uab, DataSource ds, int studyId,String detailedNotes,int discrepancyNoteTypeId) {
+    	
+        DiscrepancyNoteBean note = new DiscrepancyNoteBean();       
+        note.setDescription(message);
+        note.setDetailedNotes(detailedNotes);
+        note.setOwner(uab);
+        note.setCreatedDate(new Date());
+        note.setResolutionStatusId(ResolutionStatus.OPEN.getId());
+        note.setDiscrepancyNoteTypeId(discrepancyNoteTypeId);
+        if (parentId != null) {
+            note.setParentDnId(parentId);
+        }
+
+        note.setField(itemName);
+        note.setStudyId(studyId);
+        note.setEntityName(itemName);
+        note.setEntityType("ItemData");
+        note.setEntityValue(itemData.getValue());
+
+        note.setEventName(eventCrf.getStudyEvent().getStudyEventDefinition().getName());
+        note.setEventStart(eventCrf.getDateCreated());
+       
+        StudySubjectDAO ssdao = new StudySubjectDAO(ds);
+        StudySubjectBean ss = (StudySubjectBean) ssdao.findByPK(eventCrf.getStudySubject().getStudySubjectId());
+        note.setSubjectName(ss.getName());
+
+        note.setEntityId(itemData.getItemDataId());
+        note.setColumn("value");
+
+        DiscrepancyNoteDAO dndao = new DiscrepancyNoteDAO(ds);
+        note = (DiscrepancyNoteBean) dndao.create(note);      
+        dndao.createMapping(note);
+      
+        return note;
+    }
+
```

Comment:
As I mentioned previously, for new implementations, do not use JDBC , use Hibernate.
Also, we have other service that we use for queries, QueryService.java . use this instead.

Reply:
did code changes based on code review